File: maximum-subarray/maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        l = len(nums)
        temp = 0
        max1 =-inf
        for i in range(l):
                temp+=nums[i]
                max1=max(max1,temp)
                if temp<0:
                     temp=0
        return max1

File: maximum-subarray/maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int max_so_far=Integer.MIN_VALUE;
        int max_end=0;
        int n=nums.length;
        for(int i=0;i<n;i++){
            max_end = max_end + nums[i];
            if(max_end < nums[i]){
                max_end = nums[i];
            }
            if(max_so_far<max_end){
                max_so_far = max_end;
            }
        }
        return max_so_far;
        
    }
}